### PR TITLE
BST-11165: add license type & rule to boost-sca

### PR DESCRIPTION
**Pull request by ledo01**
This scanner can emit license finding. As such, it should be marked as a license scanner and have the rule for forbidden license usage.

Resolves: [BST-11165], [BST-11167]

[BST-11165]: https://boostsecurity.atlassian.net/browse/BST-11165?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[BST-11167]: https://boostsecurity.atlassian.net/browse/BST-11167?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by JonathanSerafini — DISMISSED** [review]

File: server-side-scanners/boostsecurityio/sbom-sca/rules.yaml [hunk 1]
```diff
@@ -15,3 +15,13 @@ rules:
     group: top10-vulnerable-components
     pretty_name: Dependency with known malicious behaviour
     ref: https://github.com/ossf/malicious-packages/tree/main/osv/malicious
+  forbidden-license:
```
Comment (lindycoder) [inline server-side-scanners/boostsecurityio/sbom-sca/rules.yaml:18]: I thought Stephan wanted to import from the license scanner

Comment (ledo01) [reply server-side-scanners/boostsecurityio/sbom-sca/rules.yaml:18]: Good point. I did the import from other scanners feature too 🤦

Comment (ledo01) [reply server-side-scanners/boostsecurityio/sbom-sca/rules.yaml]: mmh, yeah I don't think import works for server-side scanners... I don't remember it this is a bug or a restriction that we wanted. I'll check the action quickly. If there's no easy solution, I can create a rules-realm for it or just copy the rules for now.

Comment (ledo01) [reply server-side-scanners/boostsecurityio/sbom-sca/rules.yaml]: I was able to update the action to allow imports from another server-side scanner: https://github.com/boostsecurityio/scanner-registry-action/pull/48

**Review by lindycoder — APPROVED** [review]

**Review by lindycoder — COMMENTED** [review]
Whoops, wrong scanner

File: scanners/boostsecurityio/boost-sca/rules.yaml [hunk 2]
```diff
@@ -1,5 +1,6 @@
 import:
   - boostsecurityio/sca-cve
+  - boostsecurityio/oss-license
 
 rules:
   dependency-with-malicious-behaviour:
```
Comment (lindycoder) [inline scanners/boostsecurityio/boost-sca/rules.yaml:6]: Wondering if this should be only 1 place as well

**Review by lindycoder — APPROVED** [review]

File: scanners/boostsecurityio/boost-sca/rules.yaml (hunk 2 above)
Comment (ledo01) [reply scanners/boostsecurityio/boost-sca/rules.yaml:6]: Probably, can be cleaned up in a later PR.

Comment (ledo01) [reply scanners/boostsecurityio/boost-sca/rules.yaml:6]: https://github.com/boostsecurityio/dev-registry/pull/133

**Pull request merged**